src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1306,16 +1318,35 @@ private static void ComputeDeclarationsInNode(SemanticModel semanticModel, ISymb
                         executableCodeBlocks = declInNode.ExecutableCodeBlocks;
 
                         // Execute operation actions.
-                        if (shouldExecuteOperationActions && executableCodeBlocks.Any())
+                        if ((shouldExecuteOperationActions || shouldExecuteOperationBlockActions) && executableCodeBlocks.Any())
                         {
-                            var operationsToAnalyze = GetOperationsToAnalyze(executableCodeBlocks, semanticModel, cancellationToken);
-                            foreach (var analyzer in analysisScope.Analyzers)
+                            var operationBlocksToAnalyze = GetOperationBlocksToAnalyze(executableCodeBlocks, semanticModel, cancellationToken);
```

Comment:
GetOperationsToAnalyze is expensive for large methods. The computation of syntax nodes to analyze for a code block is similarly expensive, and we're doing that an equivalent number of times (in ExecuteCodeBlockActions). We should probably look into caching both the syntax nodes and the operations as Manish suggests, in a separate change set. (The performance implications of keeping around large arrays vs. recomputing them are subtle enough to debate separately.)